File: Practica 3/Cola.java
public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;
    private int elementos;

    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = -1;
        this.elementos = 0;
    }

    public void insert(long e) {
        if (!isFull()) {
            fin = (fin + 1) % n;
            arreglo[fin] = e;
            elementos++;
        } else {
            System.out.println("La cola está llena");
        }
    }

    public long remove() {
        if (!isEmpty()) {
            long valor = arreglo[inicio];
            inicio = (inicio + 1) % n;
            elementos--;
            return valor;
        } else {
            System.out.println("La cola está vacía");
            return -1;
        }
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[inicio];
        } else {
            System.out.println("La cola está vacía");
            return -1;
        }
    }

    public boolean isEmpty() {
        return elementos == 0;
    }

    public boolean isFull() {
        return elementos == n;
    }

    public int size() {
        return elementos;
    }
}

File: Practica 3/Cola.py
class Cola:
    def __init__(self, n: int):
        self.n = n
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = -1
        self.elementos = 0

    def insert(self, e: int):
        if not self.isFull():
            self.fin = (self.fin + 1) % self.n
            self.arreglo[self.fin] = e
            self.elementos += 1
        else:
            print("La cola está llena")

    def remove(self) -> int:
        if not self.isEmpty():
            valor = self.arreglo[self.inicio]
            self.inicio = (self.inicio + 1) % self.n
            self.elementos -= 1
            return valor
        else:
            print("La cola está vacía")
            return -1

    def peek(self) -> int:
        if not self.isEmpty():
            return self.arreglo[self.inicio]
        else:
            print("La cola está vacía")
            return -1

    def isEmpty(self) -> bool:
        return self.elementos == 0

    def isFull(self) -> bool:
        return self.elementos == self.n

    def size(self) -> int:
        return self.elementos
